libs/datasets/sources/zeros_filter.py

```diff
@@ -0,0 +1,48 @@
+from typing import Collection
+
+import structlog
+from covidactnow.datapublic.common_fields import CommonFields
+from covidactnow.datapublic.common_fields import PdFields
+
+from libs.datasets import timeseries
+import pandas as pd
+
+
+_log = structlog.get_logger()
+
+
+DROPPING_TIMESERIES_WITH_ONLY_ZEROS = "Dropping timeseries with only zeros"
+
+
+def drop_all_zero_timeseries(
+    ds_in: timeseries.MultiRegionDataset, fields: Collection[CommonFields]
+) -> timeseries.MultiRegionDataset:
+    ts_wide = ds_in.timeseries_wide_dates()
+
+    # Separate into timeseries in `fields` and all others.
+    variable_mask = ts_wide.index.get_level_values(PdFields.VARIABLE).isin(fields)
+    ts_wide_other_variables = ts_wide.loc[~variable_mask]
+    ts_wide_variables = ts_wide.loc[variable_mask]
+
+    # Keep rows/timeseries that have at least one value that is not 0 or NA
+    to_keep_mask = ts_wide_variables.replace(pd.NA, 0).any(axis=1)
+    to_drop = ts_wide_variables.loc[~to_keep_mask].index
+    if not to_drop.empty:
+        # Maybe add filtering to not log about the known bad data in OH counties and Loving
+        # County Texas using a RegionMask(level=County, state=OH) and some kind of RegionMask
```

Comment:
heh, loving county texas is a good example of a county that I wouldn't be surprised if the 0s were correct.  It's so small, that reporting 0s could very well be right for now.

Reply:
Yes, they haven't had enough days since first doses to do second yet:
```
iso1:us#iso2:us-tx#fips:48301,vaccinations_completed,CANScrapersStateProviders,,0,0,0,0,0,0,0,0,0,0,0,0,0,0
iso1:us#iso2:us-tx#fips:48301,vaccinations_initiated,CANScrapersStateProviders,,19,19,19,18,16,11,8,7,6,6,6,5,5,5
iso1:us#iso2:us-tx#fips:48301,vaccines_administered,CANScrapersStateProviders,,19,19,19,18,16,11,8,7,6,6,6,5,5,5
```
Added a comment.